### DISCUSSION TITLE
iPLAssからの外部データベースへの更新閲覧機能の開発について

### DISCUSSION THREAD

**Discussion by yamada1227** [dentsusoken/iPLAss#1052] [Q&A] [answered]
社内の開発効率を上げるツールを探しています。
iPLAssでは、外部データベースとの連携は可能でしょうか？
社内で開発したシステムのデータベースに接続してテーブル検索したり、マスターテーブルを更新することは可能でしょうか？
社内で使用しているデータベースは、OracleとMySQLです。
ご教示ください。
よろしくお願いいたします。

**Comment by HiguchiKiyoshi** [comment, marked as answer]
@yamada1227 
iPLAssにご関心を寄せていただき、ありがとうございます。
ご回答が遅れてしまい申し訳ございません。
ご質問いただきました件に関しまして回答させていただきます。

iPLAssの対応するデータベースに関しましてはOracle、MySQLとも対応しておりますが、
iPLAssのデータベース上におけるデータの保持の仕方は、柔軟性・拡張性を確保するため少々特殊な構造になっておりまして、
既存のシステムで設計・作成されたテーブルに対して、iPLAssが自動生成するデータ操作画面上から直接検索したり更新したりすることは難しいと思われます。

外部システムと連携する際には以下の方式とさせていただくことが多いです。

- WebAPIによる連携
https://iplass.org/docs/developerguide/webapi/index.html#_entity_crud_api
- CSVファイルでの連携
https://iplass.org/docs/developerguide/support/index.html#_export_2
https://iplass.org/docs/developerguide/support/index.html#_import_2
- データベースのView経由での連携
https://iplass.org/docs/developerguide/support/index.html#entity_view
※データベース上にiPLAss内のデータをわかりやすい形で参照（更新）可能なViewを作成します。ただ、あくまでiPLAss内に定義されたデータ構造を元にしますので、既存システムで定義済みのテーブル構造と同様の形にすることは難しいと思われます。
既存システム側のテーブルとのデータベース上でのデータ連携の仕組みは別途必要となる認識でおります。

**Reply by yamada1227** [reply]
@HiguchiKiyoshi 
ご回答ありがとうございます。
